src/server/game/Entities/Creature/GossipDef.cpp

```diff
@@ -462,6 +463,11 @@ void PlayerMenu::SendQuestGiverQuestDetails(Quest const* quest, ObjectGuid npcGU
         if (player && !sScriptMgr->ShouldBeRewardedWithMoneyInsteadOfExp(player))
         {
             questXp = uint32(quest->XPValue(playerLevel) * player->GetQuestRate());
+
+            // handle SPELL_AURA_MOD_XP_QUEST_PCT auras
+            Unit::AuraEffectList const& ModXPPctAuras = player->GetAuraEffectsByType(SPELL_AURA_MOD_XP_QUEST_PCT);
+            for (Unit::AuraEffectList::const_iterator i = ModXPPctAuras.begin(); i != ModXPPctAuras.end(); ++i)
+                AddPct(questXp, (*i)->GetAmount());
```

Comment:
Instead of reusing the same code 3 times in the core (counting Player.cpp) I would prefer if we make a function out of it and just call it in 3 places instead

Reply:
Good point, addressed this in e81febc4ed50b17636ddcc3dd8ee0f59742d5b84